cpp/src/arrow/util/future.h

```diff
@@ -493,6 +493,22 @@ class ARROW_MUST_USE_TYPE Future {
     });
   }
```

Comment:
For completeness, we could also support construction from ValueType:

```suggestion
  }
  
  /// \brief Implicit constructor to create a finished future from a value
  Future(ValueType val) : Future() {  // NOLINT runtime/explicit
    impl_ = FutureImpl::MakeFinished(FutureState::SUCCESS);
    SetResult(std::move(val));
  }
```